apps/freeablo/faworld/behaviour.cpp

```diff
@@ -38,14 +38,22 @@ namespace FAWorld
     {
         Tick ticksPassed = World::get()->getCurrentTick();
 
-        if (!mActor->isDead()) {
+        if (!mActor->isDead()) 
+        {
             const Player * nearest = FAWorld::findNearestPlayer(mActor);
-            // we are close enough to engage the player
-            if (FAWorld::squaredDistance(nearest->mPos, mActor->mPos) <= 25) {
+
+            int32_t dist = FAWorld::squaredDistance(nearest->mPos, mActor->mPos);
+
+            if (dist <= 25) // we are close enough to engage the player
+            {
                 mActor->mDestination = mActor->mPos.mGoal = nearest->mPos.current();
             }
-            // if no player is in sight, let's wander around a bit
-            else if (!mActor->mPos.mMoving) {
+            else if (dist >= 200) // just freeze if we're miles away from anyone
```

Comment:
This is a radius of about 14 tiles, which might not be enough on large enough resolutions.

For the future maybe we could extract a method into the world class or somewhere that would check if a tile is "visible" by any player and then we could use that.  This is more like just a todo note than anything, it's fine for now.